Source/WebGPU/WebGPU/CommandEncoder.mm

```diff
@@ -129,6 +129,12 @@ static MTLStoreAction storeAction(WGPUStoreOp storeOp)
     return true;
 }
 
+static void addTextureStorageIfNeeded(TextureView& texture, WGPUStoreOp storeOp, WGPULoadOp loadOp)
+{
+    if (storeOp == WGPUStoreOp_Store || loadOp == WGPULoadOp_Load)
```

Comment:
It is hard to say how web developers will use WebGPU, but if logically they specify a store depth operation because the color attachment has a store operation, they will unfortunately hit this path even if they never read from their depth texture.

It only applies to textures which only have `MTLTextureUsageRenderTarget` and not `MTLTextureUsageShader*` so if they use this render target as an attachment in a forward renderer multipass algorithm, then they will never hit this path

Reply:
oh but this check is missing the memoryless part, that happens inside `recreateBackingWithStorageIfNeeded`, so `recreateBackingWithStorageIfNeeded` may get called unnecessary but it would immediately return. I'm not sure if that is a concern